.github/workflows/colcon.yml

```diff
@@ -0,0 +1,200 @@
+      - 'Tools/scripts/**.py'
+      - 'Tools/scripts/build_sizes/**'
+      - 'Tools/scripts/build_tests/**'
+      - 'Tools/scripts/CAN/**'
+      - 'Tools/scripts/signing/**'
+      # Remove autotests stuff
+      - 'Tools/autotest/**'
+      # Remove markdown files as irrelevant
+      - '**.md'
+      # Remove dotfile at root directory
+      - './.dir-locals.el'
+      - './.dockerignore'
+      - './.editorconfig'
+      - './.flake8'
+      - './.gitattributes'
+      - './.github'
+      - './.gitignore'
+      - './.pre-commit-config.yaml'
+      - './.pydevproject'
+      - './.valgrind-suppressions'
+      - './.valgrindrc'
+      - 'Dockerfile'
+      - 'Vagrantfile'
+      - 'Makefile'
+      # Remove some directories check
+      - '.vscode/**'
+      - '.github/ISSUE_TEMPLATE/**'
+      # Remove change on other workflows
+      - '.github/workflows/test_environment.yml'
+
+  workflow_dispatch:
+
+concurrency:
+  group: ci-${{github.workflow}}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-test:
+    runs-on: ubuntu-22.04
+    container:
+      image: ardupilot/ardupilot-dev-ros:latest
+    strategy:
+      fail-fast: false  # don't cancel if a job from the matrix fails
+    steps:
+      # git checkout the PR
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          path: src/ardupilot
+      # Put ccache into github cache for faster build
+      - name: Prepare ccache timestamp
+        id: ccache_cache_timestamp
+        run: |
+          NOW=$(date -u +"%F-%T")
+          echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
+      - name: ccache cache files
+        uses: actions/cache@v3
+        with:
+          path: ~/.ccache
+          key: ${{github.workflow}}-ccache-${{steps.ccache_cache_timestamp.outputs.timestamp}}
+          restore-keys: ${{github.workflow}}-ccache-  # restore ccache from either previous build on this branch or on master
+      - name: setup ccache
+        run: |
+          . src/ardupilot/.github/workflows/ccache.env
+      # https://ardupilot.org/dev/docs/ros2.html#installation-ubuntu
+      - name: Pull in repos with vcs
+        run: |
+          cd src
+          wget https://raw.githubusercontent.com/ArduPilot/ardupilot/master/Tools/ros2/ros2.repos
+          vcs import --recursive --debug --shallow --skip-existing < ros2.repos
+      - name: Install dependencies
+        shell: 'bash'
+        run: |
+          apt update
+          rosdep update
+          # Workaround for flake8 attribute error
+          # https://github.com/ArduPilot/ardupilot/pull/24277#issuecomment-1632833433
+          python -m pip install flake8==3.7.9
+          source /opt/ros/humble/setup.bash
+          rosdep install --from-paths src --ignore-src --default-yes
+          # Install these specific versions just to prevent rolling updates.
+          # These are the latest available
+          python3 -m pip install MAVProxy==1.8.67 pymavlink==2.4.40
```

Comment:
install mavproxy.

Then, 
cd ardupilot/module/mavlink/pymavlink && sudo pip install . -U --user

Reply:
Done. I split it in two stages to make use of `working-directory`